### PR TITLE
save user in Postgres on signup

### PR DESCRIPTION
**Pull request by SimonLab**
see #41  
Create structure of the user database, we only save the github username has we don't need more information at the moment. Create migration which build the Postgres table.
If the Github authentication is successful we are saving the username in Postgres only if this one doesn't exists yet.

### CODE REVIEW

File: test/controllers/auth_controller_test.exs [hunk 1]
```diff
@@ -7,8 +7,8 @@ defmodule Dwylbot.AuthControllerTest do
     conn = get conn, "/auth/github/callback"
 
     assert html_response(conn, 302) =~ "redirected"
-    assert get_session(conn, :current_user)
-    assert get_flash(conn, :info) =~ "sucessfully authenticated"
+    assert get_session(conn, :user)
+    assert get_flash(conn, :info) =~ "Sucessfully signup, welcome to dwylbot!"
```
Comment (samhstn) [inline test/controllers/auth_controller_test.exs]: "Successful signup, welcome to dwylbot!" would be more grammatically correct

File: web/controllers/auth_controller.ex [hunk 2]
```diff
@@ -11,16 +11,19 @@ defmodule Dwylbot.AuthController do
...
   end
 
   def callback(%{assigns: %{ueberauth_failure: _fails}} = conn, _params) do
     conn
     |> put_flash(:error, "Error while authenticating")
     |> redirect(to: "/")
   end
+
+  defp basic_info(auth) do
+    %{uid: username, credentials: %{token: token}} = auth
```
Comment (samhstn) [inline web/controllers/auth_controller.ex]:
```elixir
defp basic_info(%{uid: username, credentials: %{token: token}}) do
  %{username: username, token: token}
end
```
Similar to above

File: web/models/user.ex [hunk 3]
```diff
@@ -0,0 +1,19 @@
+defmodule Dwylbot.User do
+  @moduledoc """
+  Provide functions to represent users in Postgres with Ecto
+  """
```
Comment (samhstn) [inline web/models/user.ex:4]: 👍  Nice

**Review by samhstn — CHANGES_REQUESTED** [review]
Good 👍 , reassign when ready

**Comment by nelsonic** [comment]
@SimonLab I'm _unassigning_ till @shouston3 is happy. 👍

**Review by samhstn — APPROVED** [review]

**Pull request merged**